src/intersections/rays.jl

```diff
@@ -149,58 +152,58 @@ function intersection(f, ray::Ray{3,T}, tri::Triangle{3,T}) where {T}
   det = e₁ ⋅ p
 
   # keep det > 0, modify T accordingly
-  if det > atol(T)
+  if det > atol(det)
     τ = o - vs[1]
   else
     τ = vs[1] - o
     det = -det
   end
 
-  if det < atol(T)
+  if det < atol(det)
     # This ray is parallel to the plane of the triangle.
     return @IT NotIntersecting nothing f
   end
 
   # calculate u parameter and test bounds
   u = τ ⋅ p
-  if u < -atol(T) || u > det
+  if u < -atol(u) || u > det
     return @IT NotIntersecting nothing f
   end
 
   q = τ × e₁
 
   # calculate v parameter and test bounds
   v = d ⋅ q
-  if v < -atol(T) || u + v > det
+  if v < -atol(v) || u + v > det
     return @IT NotIntersecting nothing f
   end
 
-  λ = (e₂ ⋅ q) * (one(T) / det)
+  λ = (e₂ ⋅ q) / det
 
-  if λ < -atol(T)
+  if λ < -atol(λ)
     return @IT NotIntersecting nothing f
   end
 
   # assemble barycentric weights
-  w = Vec(u, v, det - u - v)
+  w = (u, v, det - u - v)
 
-  if any(isapprox.(o, vs, atol=atol(T)))
+  if any(isapprox.(o, vs))
```

Comment:
double check if atol is passed correctly by default. we need to decide if we will pass atol explicitly in call sites or if we will fix it in the method definition of Point and Geometry in general.

Reply:
this also made me think that we can benefit from internal helper functions `isapproxequal` and `isapproxzero`, which would always use the correct atol:

```julia
isapproxequal(x, y) = isapprox(x, y, atol=atol(x))
isapproxzero(x) = isapprox(x, zero(x), atol=atol(x))
```

That way we save ourselves from copying pasting atol explicitly everywhere.